src/Illuminate/Redis/Connections/PhpRedisConnection.php

```diff
@@ -495,17 +494,17 @@ public function createSubscription($channels, Closure $callback, $method = 'subs
     /**
      * Flush the selected Redis database.
      *
-     * @return void
+     * @return mixed
      */
     public function flushdb()
     {
-        if (! $this->client instanceof RedisCluster) {
-            return $this->command('flushdb');
-        }
+        $arguments = func_get_args();
 
-        foreach ($this->client->_masters() as $master) {
-            $this->client->flushDb($master);
+        if (strtoupper((string) $arguments[0] ?? null) === 'ASYNC') {
```

Comment:
This causes a lot of tests to fail for me.

It should be `if (strtoupper((string) ($arguments[0] ?? null)) === 'ASYNC') {`

Reply:
As sometimes there are no arguments at all.

Reply:
@tillkruss ^

Reply:
I'll make a PR with tests and the changes.

Reply:
See https://github.com/laravel/framework/pull/40571